### PR TITLE
feat(node): Integrate bitswap protocol for shwap

### PR DESCRIPTION
**Pull request by oblique**

### CODE REVIEW

File: node/src/p2p.rs [hunk 1]
```diff
@@ -393,6 +468,7 @@ where
     S: Store + 'static,
 {
     autonat: autonat::Behaviour,
+    bitswap: BitswapBehaviour<64, InMemoryBlockstore<64>>,
```
Comment (fl0rek) [inline node/src/p2p.rs:471]: Could you define const for 64 and swap there everywhere?

File: node/src/p2p.rs [hunk 2]
```diff
@@ -106,6 +123,18 @@ pub enum P2pError {
...
     #[error("Bootnode multiaddrs without peer ID: {0:?}")]
     BootnodeAddrsWithoutPeerId(Vec<Multiaddr>),
+
+    /// An error propagated from [`beetswap::Behaviour`].
+    #[error("Bitswap: {0}")]
+    Bitswap(#[from] beetswap::Error),
+
+    /// ProtoBuf message failed to be decoded.
+    #[error("ProtoBuf decoding error: {0}")]
+    ProtoDecodeFailed(celestia_tendermint_proto::Error),
```
Comment (zvolin) [inline node/src/p2p.rs:133]: any reason to not use `#[from]` there?

Comment (zvolin) [reply node/src/p2p.rs:133]: and `Ok(foo::decode(...)?)` instead of map_err

**Review by zvolin — APPROVED** [review]
looks good, left some nit

File: node/src/p2p.rs (hunk 2 above)
Comment (oblique) [reply node/src/p2p.rs]: no reason. added it

**Review by fl0rek — APPROVED** [review]
👌

**Pull request merged**